src/Site_Command.php

```diff
@@ -309,6 +341,210 @@ public function info( $args ) {
+	}
+
+	/**
+	 * Stops containers associated with site.
+	 *
+	 * <site-name>
+	 * : Name of the site.
+	 *
+	 * [--all]
+	 * : Stop all containers of site.
+	 *
+	 * [--nginx]
+	 * : Stop nginx container of site.
+	 *
+	 * [--php]
+	 * : Stop php container of site.
+	 *
+	 * [--mysql]
+	 * : Stop mysql container of site.
+	 *
+	 * [--redis]
+	 * : Stop redis container of site.
+	 *
+	 * [--mailhog]
+	 * : Stop mailhog container of site.
+	 *
+	 * [--phpmyadmin]
+	 * : Stop phpmyadmin container of site.
+	 *
+	 * [--phpredisadmin]
+	 * : Stop phpredisadmin container of site.
+	 *
+	 * [--adminer]
+	 * : Stop adminer container of site.
+	 *
+	 * [--anemometer]
+	 * : Stop anemometer container of site.
+	 */
+	public function stop( $args, $assoc_args ) {
+		$this->site_docker_compose_execute( $args[0], 'stop', $args, $assoc_args);
+	}
+
+	/**
+	 * Restarts containers associated with site.
+	 *
+	 * <site-name>
+	 * : Name of the site.
+	 *
+	 * [--all]
+	 * : Restart all containers of site.
+	 *
+	 * [--nginx]
+	 * : Restart nginx container of site.
+	 *
+	 * [--php]
+	 * : Restart php container of site.
+	 *
+	 * [--mysql]
+	 * : Restart mysql container of site.
+	 *
+	 * [--redis]
+	 * : Restart redis container of site.
+	 *
+	 * [--mailhog]
+	 * : Restart mailhog container of site.
+	 *
+	 * [--phpmyadmin]
+	 * : Restart phpmyadmin container of site.
+	 *
+	 * [--phpredisadmin]
+	 * : Restart phpredisadmin container of site.
+	 *
+	 * [--adminer]
+	 * : Restart adminer container of site.
+	 *
+	 * [--anemometer]
+	 * : Restart anemometer container of site.
+	 */
+	public function restart( $args, $assoc_args ) {
+		$this->site_docker_compose_execute( $args[0], 'restart', $args, $assoc_args);
+	}
+
+	/**
+	 * Reload services in containers without restarting container(s) associated with site.
+	 *
+	 * <site-name>
+	 * : Name of the site.
+	 *
+	 * [--all]
+	 * : Reload all services of site(which are supported).
+	 *
+	 * [--nginx]
+	 * : Reload nginx service in container.
+	 *
+	 * [--php]
+	 * : Start php service in container.
+	 */
+	public function reload( $args, $assoc_args ) {
+		$this->site_docker_compose_execute( $args[0], 'reload', $args, $assoc_args);
+	}
+
+	private function site_docker_compose_execute( $site, $action, $args, $assoc_args ) {
+		$all = \EE\Utils\get_flag_value( $assoc_args, 'all' );
+		$no_service_specified = count( $assoc_args ) === 0 ;
+
+		$this->populate_site_info( $args );
+
+		chdir( $this->site_root );
+
+		if( $all || $no_service_specified ) {
+			if( $action === 'reload' ) {
+				$this->reload_services( [ 'nginx', 'php' ] );
+				return;
+			}
+			$this->run_compose_command( $action, '', null, 'all services' );
+		}
+		else {
+			$services = array_map( [$this, 'map_args_to_service'], array_keys( $assoc_args ) );
+
+			if( $action === 'reload' ) {
+				$this->reload_services( $services );
+				return;
+			}
+
+			foreach( $services as $service ) {
```

Comment:
I think we should add some white-list check on `$action` if this actions is not from reload/restart/stop/start etc. then do not perform anything and add error message.

Reply:
You can do this same for `$service` also.

Reply:
I don't think that whitelist option is needed as we aren't supplying untrusted input directly into the commands. We are statically adding them.

Reply:
yes, sorry my bad. It's already white-listed on command doc block.